### PR TITLE
Add client UX for redirecting to an external portal for synced accounts

### LINKED ISSUE #1713: UX: If synced from ext src, refer users there for self-service
**Is your feature request related to a problem? Please describe.**

If Kani gets its user data/credentials from LDAP or another external source, local changes to synced attributes are blocked. Users might be unaware about where to self-service update their credentials instead.

**Describe the solution you'd like**

If users try to self-service update their attributes/credentials, they should be referred to the actual self-service method of the Kanidm instance's sync "source of truth".

**Describe alternatives you've considered**

n/a

**Additional context**

Discussed beforehand and @Firstyear told me to create an issue, so it doesn't get forgotten.

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes #1713  - This is a WIP. 

- [ ] This pr contains no AI generated code
- [ ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
@yaleman and @ziQ - I want to check something with you about access controls of the sync account credential portal url.

When an account is synced, it needs to be able to get the credential portal url by itself in the ui flows. This means we have to pass through access controls.

There are two possible ways to approach this:

* Allow all users including anonymous to read the value of sync_credential_portal
* Limit the sync_credential_portal value to just the synchronsied users.

IMO the url isn't private info, but I can also see why some deployments way wish to protect this. I think if we want to limit it to synced users only we can do similar to what we did with oauth with an access control overlay such that only synced users of the same sync domain can read the attribute in question. 

Thoughts?

File: server/lib/src/constants/acp.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ lazy_static! {
             Value::new_json_filter_s("{\"eq\": [\"class\", \"recycled\"]}")
                 .expect("Invalid JSON filter")
         ),
+        ("acp_search_attr", Value::new_iutf8("class")),
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply server/lib/src/constants/acp.rs:26]: Yes - without it queries don't work, so each search rule needs the class search.

[the remark this answers is not in the document]
Comment (Firstyear) [reply server/lib/src/constants/acp.rs:26]: Oh this one actualy is a dupe.

**Comment by yaleman** [comment]
> There are two possible ways to approach this:
> 
> * Allow all users including anonymous to read the value of sync_credential_portal
> * Limit the sync_credential_portal value to just the synchronsied users.
> 
> IMO the url isn't private info, but I can also see why some deployments way wish to protect this. I think if we want to limit it to synced users only we can do similar to what we did with oauth with an access control overlay such that only synced users of the same sync domain can read the attribute in question.

I'd make it only the sync'd users (and admins/service desk) out of an abundance of caution TBH.

(Sorry, somehow I didn't see your comment above... weird)

**Comment by Firstyear** [comment]
CLI:

```
spn: ipatestuser@localhost
Name: ipa test user full name
Externally Managed: Contact your admin to update your account details.
Primary Credential:
  unable to modify
Passkeys:
  unable to modify
Can Commit: true

spn: ipatestuser@localhost
Name: ipa test user full name
Externally Managed: Visit https://sink.ipa.blackhats.net.au/ipa/ui/#/e/user/search to update your account details.
Primary Credential:
  unable to modify
Passkeys:
  unable to modify
Can Commit: true

spn: testuser@localhost
Name: TestUser
Primary Credential:
  not set
Passkeys:
  not set
Can Commit: true
```

**Comment by Firstyear** [comment]
Might just add one more test ...

**Review by yaleman — APPROVED** [review]

**Pull request merged**